File: Week 2/Assignment 6/problem1.java

public class problem1 {
 
    public static int findMinRec(int A[], int n)
    {
      if(n == 1)
        return A[0];
         
        return Math.min(A[n-1], findMinRec(A, n-1));
      
    }
    public static int findMaxRec(int A[], int n){
        if(n == 1)
        return A[0];
         
        return Math.max(A[n-1], findMaxRec(A, n-1));
    }
     
    public static void main(String args[])
    {
        int A[] = {1,-1,0,2,-2,3,-3,4,-4};
        int n = A.length;
         
        System.out.println(findMinRec(A, n));
        System.out.println(findMaxRec(A, n));
    }
}

File: February/Week 1/Assignment 7/problem1.js
let max=(array)=> {
    if (array.length === 1) {  // Step1: set up your base case
        return array[0]
   } else {  
       return Math.max(array.shift(), max(array)) // Step2: rec case
   }
  }

  let min=(array)=> {
    if (array.length === 1) {  // Step1: set up your base case
        return array[0]
   } else {  
       return Math.min(array.shift(), min(array)) // Step2: rec case
   }
  }  

console.log(max([1,-1,0,2,-2,3,-3,4,-4]))
console.log(min([1,-1,0,2,-2,3,-3,4,-4]))